assets/src/js/admin/reports/components/list/index.js

```diff
@@ -1,6 +1,35 @@
-const List = (props) => {
+import { useEffect, createRef } from 'react'
+import './style.scss';
+
+const List = ({onScrollEnd, children}) => {
+
+    const list = createRef()
+
+    useEffect(() => {
+
+        function checkScroll (evt) {
+
+            const remaining = evt.target.scrollHeight - evt.target.scrollTop
+            const height = evt.target.offsetHeight
+
+            if (remaining <= height) {
+                onScrollEnd()
+            }
+
+        }
+        
+        list.current.addEventListener('scroll', checkScroll)
+    
+        return function cleanup () {
+            list.current.removeEventListener('scroll', checkScroll)
+        }
```

Comment:
I suggest allowing `onScrollEnd` to be optional. As such, only construct and destruct the event listener if that variable is set.

Reply:
done! I added a conditional check around this code to first check that onScrollEnd is defined